tests/e2e/utils/slack.ts

```diff
@@ -0,0 +1,161 @@
+
+/**
+ * External dependencies
+ */
+import { createReadStream } from 'fs';
+import {
+	WebClient,
+	ErrorCode,
+	CodedError,
+	WebAPICallResult,
+} from '@slack/web-api';
+
+/**
+ * This integration was adapted from @woocommerce/e2e-environment.
+ * https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/e2e-environment/src/slack/reporter.js
+ */
+
+const {
+	GITHUB_ACTIONS,
+	GITHUB_REF,
+	GITHUB_SHA,
+	GITHUB_REPOSITORY,
+	GITHUB_RUN_ID,
+	E2E_SLACK_TOKEN,
+	E2E_SLACK_CHANNEL_ID,
+	WC_E2E_SCREENSHOTS,
+} = process.env;
+
+interface SlackParams {
+	branch: string;
+	commit: string;
+	webUrl?: string;
+}
+
+interface ErrorWithData extends CodedError {
+	data: WebAPICallResult & { error: string };
+}
+
+const errorsWithData = [
+	ErrorCode.FileUploadInvalidArgumentsError,
+	ErrorCode.PlatformError,
+];
+
+let web: WebClient;
+
+/**
+ * Initialize the Slack web client.
+ */
+const initializeWeb = (): WebClient => {
+	if ( ! web ) {
+		web = new WebClient( E2E_SLACK_TOKEN );
+	}
+	return web;
+};
+
+/**
+ * Initialize Slack parameters if tests are running in CI.
+ */
+const initializeSlack = (): SlackParams | undefined => {
+	if ( ! WC_E2E_SCREENSHOTS || ! E2E_SLACK_TOKEN ) {
+		return;
+	}
+
+	if ( ! GITHUB_ACTIONS ) {
+		return {
+			branch: 'local environment',
+			commit: 'latest',
+		};
+	}
+
+	// Build PR info
+	const refArray = GITHUB_REF.split( '/' );
+	const branch = refArray.pop();
+
+	return {
+		branch,
+		commit: GITHUB_SHA,
+		webUrl: `https://github.com/${ GITHUB_REPOSITORY }/actions/runs/${ GITHUB_RUN_ID }`,
+	};
+};
+
+const handleRequestError = (
+	error: unknown,
+	errorMessage = 'An unexpected error occurred'
+) => {
+	if ( errorsWithData.includes( ( error as CodedError ).code ) ) {
+		console.log( errorMessage, ( error as ErrorWithData ).data );
+	} else {
+		console.log( errorMessage, error );
+	}
+};
+
+/**
+ * Post a message to a Slack channel for a failed test.
+ */
+export const sendFailedTestMessageToSlack = async ( testName: string ) => {
+	const slackParams = initializeSlack();
+
+	if ( ! slackParams?.branch ) {
+		return;
+	}
+
+	const { branch, commit, webUrl } = slackParams;
+	const webClient = initializeWeb();
+
+	try {
+		// Adding the app does not add the app user to the channel
+		await webClient.conversations.join( {
+			channel: E2E_SLACK_CHANNEL_ID,
+			token: E2E_SLACK_TOKEN,
+		} );
+	} catch ( error ) {
+		if ( error.data.error === 'channel_not_found' ) {
+			return;
+		}
+
+		handleRequestError( error, 'Failed to join the channel' );
+	}
+
+	try {
+		await webClient.chat.postMessage( {
+			channel: E2E_SLACK_CHANNEL_ID,
+			token: E2E_SLACK_TOKEN,
+			text: `Test failed on *${ branch }* branch. \n
+            The commit this build is testing is *${ commit }*. \n
+            The name of the test that failed: *${ testName }*. \n
+            See screenshot of the failed test below. ${
+				webUrl ? `*Build log* could be found here: ${ webUrl }` : ''
```

Comment:
Nitpick: Can we change `could` to `can`?

Reply:
Definitely! I've updated it on [d616ea1](https://github.com/Automattic/woocommerce-payments/pull/10362/commits/d616ea1209013b106c5f3d42720a09ca18def3bb). Thanks for catching it!